utils/check_roxie/check_roxie

```diff
@@ -1,11 +1,82 @@
-#!/bin/bash
+#!/usr/bin/python
 
-OUTPUT=`testsocket $1 "<control:alive/>" |grep Status |grep ok -c`
+import sys, re, os, random
+
+#Output usage if argument is missing
+try:
+        IP = sys.argv[1]
+except IndexError:
+                print "\nNeeds IP address of the target server"
+                print """
+Usage:
+        /usr/lib64/nagios/plugins/check_roxie <ip-address>
+"""
+                sys.exit(1)
+
+status = os.popen("2>/dev/null testsocket %s \"<control:alive/>\"" % IP)
+status_result = status.read()
+roxie_status = re.findall('\<Status\>(.+?)\</Status\>+?', status_result)
+servers = os.popen("2>/dev/null testsocket %s \"<control:topology/>\"" % IP)
+servers_result = servers.read()
+servers_list = re.findall('netAddress="(.*?)"/>', servers_result)
+
+#Check if roxie process is up and running
+if len(roxie_status) > 0:
+        if roxie_status[0] == 'ok':
+		#Get state hash if roxie cluster has less than 5 servers
```

Comment:
Seems a bit arbitrary to select 5 - also if there IS a state mismatch error it typically manifests as one node being out of step with all the others, so you are unlikely to spotit. Is it really that painful to request state from all? If you use -lock mode on testsocket you can get them all with a single call...